File: allgorythms/algorithms/go/sort/insertion.go
package insertion

func InsertionSort(list []int) {
	for i := 1; i < len(list); i++ {
		j := i
		for j > 0 && list[j] < list[j-1] {
			list[j], list[j-1] = list[j-1], list[j]
			j--
		}
	}
}

File: allgorythms/algorithms/cpp/sort/insertion.cpp
void insertion_sort(int s[], int n)
{
    int i, j;
    for (i=1; i<n; i++) {
        j=i;
        while ((j>0) && (s[j] < s[j-1])) {
            int tmp = s[j];
            s[j]=s[j-1];
            s[j-1] = tmp;
            j=j-1;
        }
    }
}
